### ISSUE TITLE
Compactor took 70 minutes to start work

### ISSUE THREAD

**Issue by bboreham** [cortexproject/cortex#3841]
**Describe the bug**
Compactor was stopped on one node by `kubectl drain` then moved to another node.
On startup, it did some kind of work for ~70 minutes before starting "real" work.

Compactor logs after startup:

```
level=info ts=2021-02-19T06:11:18.739889355Z caller=main.go:187 msg="Starting Cortex" version="(version=1.7.0-rc.0, branch=HEAD, revision=af2c64e)"
level=info ts=2021-02-19T06:11:18.740461621Z caller=server.go:229 http=[::]:80 grpc=[::]:9095 msg="server listening on addresses"
level=info ts=2021-02-19T06:11:18.742489959Z caller=module_service.go:58 msg=initialising module=memberlist-kv
level=info ts=2021-02-19T06:11:18.741951771Z caller=module_service.go:58 msg=initialising module=server
level=info ts=2021-02-19T06:11:18.742692682Z caller=module_service.go:58 msg=initialising module=compactor
level=info ts=2021-02-19T06:11:18.75085362Z caller=lifecycler.go:521 msg="not loading tokens from file, tokens file path is empty"
level=info ts=2021-02-19T06:11:18.751278858Z caller=compactor.go:317 component=compactor msg="waiting until compactor is ACTIVE in the ring"
level=info ts=2021-02-19T06:11:18.78962424Z caller=lifecycler.go:550 msg="instance not found in ring, adding with no tokens" ring=compactor
level=info ts=2021-02-19T06:11:18.804099438Z caller=lifecycler.go:397 msg="auto-joining cluster after timeout" ring=compactor
level=info ts=2021-02-19T06:11:20.508101255Z caller=compactor.go:331 component=compactor msg="waiting until compactor ring topology is stable" min_waiting=1m0s max_waiting=5m0s
level=info ts=2021-02-19T06:11:20.507975886Z caller=compactor.go:321 component=compactor msg="compactor is ACTIVE in the ring"
level=info ts=2021-02-19T06:12:20.508583011Z caller=blocks_cleaner.go:129 component=cleaner msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:12:20.508278927Z caller=compactor.go:335 component=compactor msg="compactor is ring topology is stable"
level=info ts=2021-02-19T06:12:24.10330928Z caller=blocks_cleaner.go:284 component=cleaner org_id=10445 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:12:53.037530823Z caller=blocks_cleaner.go:298 component=cleaner org_id=10445 msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T06:12:53.621570113Z caller=blocks_cleaner.go:284 component=cleaner org_id=10651 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:12:53.62140936Z caller=blocks_cleaner.go:289 component=cleaner org_id=10445 msg="completed blocks cleanup and maintenance" duration=29.518100292s
level=info ts=2021-02-19T06:13:09.248615252Z caller=blocks_cleaner.go:298 component=cleaner org_id=10651 msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T06:13:09.357681295Z caller=blocks_cleaner.go:289 component=cleaner org_id=10651 msg="completed blocks cleanup and maintenance" duration=15.73610944s
level=info ts=2021-02-19T06:13:09.357940217Z caller=blocks_cleaner.go:284 component=cleaner org_id=10925 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:13:20.158960937Z caller=blocks_cleaner.go:298 component=cleaner org_id=10925 msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T06:13:20.452977177Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYTNN5ADJGRWW6XQJVJMVBN5
level=info ts=2021-02-19T06:13:20.633161469Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYV1NGPA6G6S05574TQR30KF
level=info ts=2021-02-19T06:13:20.867214047Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYTG3Q7VCFVZMH0VMFWZSJH4
level=info ts=2021-02-19T06:13:21.104545233Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYV1S5K4SBKMZ94ZW2NDEW1V
level=info ts=2021-02-19T06:13:21.302032372Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYSJ6DW2PYM078TCGX03ZQ20
level=info ts=2021-02-19T06:13:21.496926951Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYT9RV0RA97WX30B0ABJBY5H
level=info ts=2021-02-19T06:13:21.689340307Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYT9RY973CED37ZMMXANZECG
level=info ts=2021-02-19T06:13:22.34341879Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYSW9F6FKWMJRVSMCSWNXG5K
level=info ts=2021-02-19T06:13:22.583427653Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYV1MB3CFDH73GXSWWPGHPDY
level=info ts=2021-02-19T06:13:22.833000871Z caller=blocks_cleaner.go:334 component=cleaner org_id=10925 msg="deleted block marked for deletion" block=01EYV1RRS5SMA81KAMHMP75XCV
level=info ts=2021-02-19T06:13:22.948749403Z caller=blocks_cleaner.go:289 component=cleaner org_id=10925 msg="completed blocks cleanup and maintenance" duration=13.59081146s
level=info ts=2021-02-19T06:13:22.948796668Z caller=blocks_cleaner.go:284 component=cleaner org_id=11491 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:13:49.137497013Z caller=blocks_cleaner.go:298 component=cleaner org_id=11491 msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T06:13:49.510571389Z caller=blocks_cleaner.go:289 component=cleaner org_id=11491 msg="completed blocks cleanup and maintenance" duration=26.561773264s
level=info ts=2021-02-19T06:13:49.510631202Z caller=blocks_cleaner.go:284 component=cleaner org_id=12035 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:14:17.563680492Z caller=blocks_cleaner.go:298 component=cleaner org_id=12035 msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T06:14:18.061502624Z caller=blocks_cleaner.go:284 component=cleaner org_id=12036 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:14:18.061452012Z caller=blocks_cleaner.go:289 component=cleaner org_id=12035 msg="completed blocks cleanup and maintenance" duration=28.550819958s
level=info ts=2021-02-19T06:14:45.919297894Z caller=blocks_cleaner.go:298 component=cleaner org_id=12036 msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T06:14:46.14217929Z caller=blocks_cleaner.go:289 component=cleaner org_id=12036 msg="completed blocks cleanup and maintenance" duration=28.080674657s
level=info ts=2021-02-19T06:14:46.14234109Z caller=blocks_cleaner.go:284 component=cleaner org_id=12291 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:15:12.33108636Z caller=blocks_cleaner.go:298 component=cleaner org_id=12291 msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T06:15:12.488251466Z caller=blocks_cleaner.go:284 component=cleaner org_id=12360 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T06:15:12.488102518Z caller=blocks_cleaner.go:289 component=cleaner org_id=12291 msg="completed blocks cleanup and maintenance" duration=26.345760867s
level=info ts=2021-02-19T06:15:35.092011653Z caller=blocks_cleaner.go:298 component=cleaner org_id=12360 msg="migrated block deletion marks to the global markers location"
...
Level=info ts=2021-02-19T07:24:17.089271804Z caller=blocks_cleaner.go:284 component=cleaner org_id=9259 msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T07:24:36.299456096Z caller=blocks_cleaner.go:298 component=cleaner org_id=9259 msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T07:24:36.773483034Z caller=blocks_cleaner.go:284 component=cleaner org_id=migration msg="started blocks cleanup and maintenance"
level=info ts=2021-02-19T07:24:36.773426433Z caller=blocks_cleaner.go:289 component=cleaner org_id=9259 msg="completed blocks cleanup and maintenance" duration=19.684153771s
level=info ts=2021-02-19T07:24:36.871757302Z caller=blocks_cleaner.go:298 component=cleaner org_id=migration msg="migrated block deletion marks to the global markers location"
level=info ts=2021-02-19T07:24:36.992835257Z caller=blocks_cleaner.go:289 component=cleaner org_id=migration msg="completed blocks cleanup and maintenance" duration=219.351143ms
level=info ts=2021-02-19T07:24:36.992887657Z caller=blocks_cleaner.go:133 component=cleaner msg="successfully completed blocks cleanup and maintenance"
level=info ts=2021-02-19T07:24:36.992988605Z caller=cortex.go:389 msg="Cortex started"
level=info ts=2021-02-19T07:24:36.99290621Z caller=compactor.go:407 component=compactor msg="discovering users from bucket"
level=info ts=2021-02-19T07:24:37.074585705Z caller=compactor.go:414 component=compactor msg="discovered users from bucket" users=229
level=info ts=2021-02-19T07:24:37.087097526Z caller=compactor.go:452 component=compactor msg="starting compaction of user blocks" user=17676
level=info ts=2021-02-19T07:24:37.105187542Z caller=compact.go:936 component=compactor org_id=17676 msg="start sync of metas"
level=info ts=2021-02-19T07:24:37.549106333Z caller=fetcher.go:458 component=compactor org_id=17676 component=block.BaseFetcher msg="successfully synchronized block metadata" duration=443.875441ms cached=131 returned=57 partial=0
level=info ts=2021-02-19T07:24:37.549162472Z caller=compact.go:941 component=compactor org_id=17676 msg="start of GC"
level=info ts=2021-02-19T07:24:37.549385777Z caller=compact.go:953 component=compactor org_id=17676 msg="start of compactions"
```

(I removed many lines saying "could not guess file size for multipart upload; upload might be not optimized")

**To Reproduce**
Unknown

**Expected behavior**
Should get going within a short time.

**Environment:**
 - Infrastructure: Kubernetes, EBS disk

**Storage Engine**
- [x] Blocks
- [ ] Chunks

Compactor args:
```
        - -blocks-storage.backend=s3
        - -blocks-storage.bucket-store.ignore-deletion-marks-delay=1h
        - -blocks-storage.bucket-store.sync-dir=/data/tsdb
        - -blocks-storage.s3.bucket-name=<redacted>
        - -blocks-storage.s3.endpoint=s3.dualstack.us-east-1.amazonaws.com
        - -blocks-storage.tsdb.block-ranges-period=2h
        - -blocks-storage.tsdb.dir=/data/tsdb
        - -blocks-storage.tsdb.retention-period=96h
        - -blocks-storage.tsdb.ship-interval=1m
        - -compactor.block-ranges=2h,12h,24h
        - -compactor.cleanup-concurrency=1
        - -compactor.compaction-concurrency=1
        - -compactor.compaction-interval=30m
        - -compactor.data-dir=/data
        - -compactor.ring.consul.hostname=consul.cortex.svc.cluster.local:8500
        - -compactor.ring.prefix=
        - -compactor.ring.store=consul
        - -compactor.sharding-enabled=true
        - -store-gateway.sharding-enabled=true
        - -store-gateway.sharding-ring.consul.hostname=consul.cortex.svc.cluster.local:8500
        - -store-gateway.sharding-ring.prefix=
        - -store-gateway.sharding-ring.replication-factor=3
        - -store-gateway.sharding-ring.store=consul
        - -store.engine=blocks
        - -target=compactor
```

Note `-compactor.cleanup-concurrency=1` which was set after #3794

**Comment by roidelapluie** [comment]
Do you think it is new in 1.7 and that it should delay the release?

**Comment by bboreham** [comment]
I have no basis to answer that; I didn't run 1.6 compactor.

**Comment by bboreham** [comment]
I notice `rate(thanos_objstore_bucket_operation_failures_total{operation="get",kubernetes_pod_name="compactor-0"}[1m])` was 2-3 per second for the whole 70 minutes.  Don't see any failures in the logs.

**Comment by bboreham** [comment]
Looking at the logs, each tenant took ~30 seconds between `"started blocks cleanup and maintenance"` and `"migrated block deletion marks to the global markers location"`.

I see that the CHANGELOG advises me I "can safely disable it via `-compactor.block-deletion-marks-migration-enabled=false`; maybe that advice should be stronger?

**Comment by pracucci** [comment]
> I see that the CHANGELOG advises me I "can safely disable it via -compactor.block-deletion-marks-migration-enabled=false; maybe that advice should be stronger?

Agree. What's about #3842?

**Comment by bboreham** [comment]
I would put it at the top of the 1.7 notes, since we need someone upgrading to 1.7 to read it, and we expect to remove this advice in 1.8.

**Comment by pracucci** [comment]
> I would put it at the top of the 1.7 notes, since we need someone upgrading to 1.7 to read it, and we expect to remove this advice in 1.8.

@khaines Can you do it please?

**Comment by bboreham** [comment]
Closed by #3857 and #3842
